src/net/tcp/listener.rs

```diff
@@ -248,3 +248,21 @@ impl FromRawFd for TcpListener {
         TcpListener::from_std(FromRawFd::from_raw_fd(fd))
     }
 }
+
+impl From<TcpListener> for net::TcpListener {
+    /// Converts a `mio::net::TcpListener` into a `std::net::TcpListener`
+    fn from(listener: TcpListener) -> Self {
+        // Safety: This is safe since we are extracting the raw fd from a well-constructed
+        // mio::net::TcpListener which ensures that we actually pass in a valid file
+        // descriptor/socket
+        unsafe {
+            #[cfg(any(unix, target_os = "hermit"))]
+            let std_listener = net::TcpListener::from_raw_fd(listener.into_raw_fd());
+
+            #[cfg(windows)]
+            let std_listener = net::TcpListener::from_raw_socket(listener.into_raw_socket());
+
+            std_listener
```

Comment:
Don't need to create a binding.
```suggestion
            #[cfg(any(unix, target_os = "hermit"))]
            net::TcpListener::from_raw_fd(listener.into_raw_fd())
            #[cfg(windows)]
            net::TcpListener::from_raw_socket(listener.into_raw_socket())
```

Reply:
My editor was complaining about a missing semicolon in that case so I added the binding. But you are right here, it compiles just fine without it so I will apply this to all trait implementations.